File: 1-Two-Sum.py
class Solution:
	def twoSum(self, nums, target):
		indexOfComponets = []
		flag = 0
		for index1 in range(len(nums)):
			if flag == 1:
				break
			for index2 in range(len(nums)):
				if (nums[index1] + nums[index2] == target) and (index1 != index2):
					indexOfComponets.append(index1)
					indexOfComponets.append(index2)
					flag = 1
		finalSolution = indexOfComponets[0:2]
		return(finalSolution)

# sol = Solution()
# array = [2,7,11,15]
# number = 9
# print(sol.twoSum(array,number))

File: 1-Two-Sum.cpp
// Problem:
// Given an array of integers, return indices of the two numbers such that they add up to a specific target.
// You may assume that each input would have exactly one solution, and you may not use the same element twice.

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        std::vector<int> sol;
        std::unordered_map<int,int> dict;
        int i = 0;
        
        for(std::vector<int>::iterator iter = nums.begin(); iter != nums.end(); ++iter)
        {
            auto lookup = dict.find(*iter);
            if(lookup != dict.end())
            {
                return(std::vector<int> {lookup->second,i});
            }
            else
            {
                dict.insert(std::pair<int,int>(target-*iter,i));
            }
            i++;
        }
        return(sol);
    }
};